src/main/java/com/selfxdsd/storage/SelfProjects.java

```diff
@@ -262,23 +239,23 @@ public Projects page(final Page page) {
     @Override
     public Iterator<Project> iterator() {
         final Page page = super.current();
-        return this.database.jooq()
+        final DSLContext jooq = this.database.jooq();
+        return jooq
             .select()
-            .from(SLF_PROJECTS_XDSD)
+            .from(jooq.select()
+                .from(SLF_PROJECTS_XDSD)
+                .limit(page.getSize())
+                .offset((page.getNumber() - 1) * page.getSize())
+                .asTable("projects_page"))
             .join(SLF_USERS_XDSD)
-            .on(
-                SLF_PROJECTS_XDSD.USERNAME.eq(SLF_USERS_XDSD.USERNAME).and(
-                    SLF_PROJECTS_XDSD.PROVIDER.eq(SLF_USERS_XDSD.PROVIDER)
-                )
-            )
+            .on(DSL.field("projects_page.username")
+                    .eq(SLF_USERS_XDSD.USERNAME)
+                    .and(DSL.field("projects_page.provider")
+                        .eq(SLF_USERS_XDSD.PROVIDER)))
             .join(SLF_PMS_XDSD)
-            .on(
-                SLF_PROJECTS_XDSD.PMID.eq(SLF_PMS_XDSD.ID)
-            )
-//            .limit(page.getSize())
```

Comment:
@amihaiemil Here was the culprit that caused showing projects in "random" order. We needed a "paged table" in iterator too. `SelfContributors#iterator` also has this bug (limit/offset at the end of join).